src/Symfony/Bridge/Twig/Translation/TwigExtractor.php

```diff
@@ -77,8 +77,37 @@ protected function extractTemplate(string $template, MessageCatalogue $catalogue
 
         $this->twig->parse($this->twig->tokenize(new Source($template, '')));
 
-        foreach ($visitor->getMessages() as $message) {
-            $catalogue->set(trim($message[0]), $this->prefix.trim($message[0]), $message[1] ?: $this->defaultDomain);
+        foreach ($visitor->getMessages() as $extractedMessage) {
+            $message = trim($extractedMessage[0]);
+            $domain = $extractedMessage[1] ?: $this->defaultDomain;
+
+            $catalogue->set($message, $this->prefix.trim($extractedMessage[0]), $domain);
+
+            if (!empty($extractedMessage[2])) {
+                $metadata = $catalogue->getMetadata($message, $domain);
+                $variablesNote = [
+                    'category' => 'symfony-extractor-variables',
+                    'content' => 'Available variables: '.implode(', ', $extractedMessage[2]),
+                ];
+
+                // Update old variables note (if any)
+                if (isset($metadata['notes'])) {
+                    foreach ($metadata['notes'] as $index => $note) {
+                        if (isset($note['category']) && 'symfony-extractor-variables' === $note['category']) {
+                            // Keep the higher variables count
+                            if (\count($extractedMessage[2]) > substr_count($note['content'], ',')) {
+                                $metadata['notes'][$index] = $variablesNote;
+                            }
```

Comment:
Hm.. is this true.. What if I change from 3 vars to 2 vars. Then the note will still show 3 vars. 

Could we be smarter? Or could we just use the data we extracted?

Reply:
Good point but I tried to take care of that case.

Take the following Twig code as an example:
```twig
{% set my_translation = t('my_message', {'var1': 'val1'}) %}
{% set my_translation = t('my_message', {'var1': 'val1', 'var2': 'val2', 'var3': 'val3'}) %}
{% set my_translation = t('my_message', {'var1': 'val1', 'var2': 'val2'}) %}
```

The command will extract: `Variables: var1, var2, var3` _(my code ensures that the third line doesn't overwrite the second one)_.

Then we change it to:
```twig
{% set my_translation = t('my_message', {'var1': 'val1'}) %}
{% set my_translation = t('my_message', {'var1': 'val1', 'var2': 'val2'}) %}
{% set my_translation = t('my_message', {'var1': 'val1', 'var2': 'val2'}) %}
```

The command will extract: `Variables: var1, var2` _(old value will be replaced since the check is performed against newly extracted messages only)_.

The point of the lines you mentioned is to ensure that the **highest** between all the newly extracted options is kept, but this doesn't include old messages that have been extracted before.

If you have any smarter behavior in mind I can try to implement that.

Reply:
Oh, I didn't even consider that. Well done. 

Im very happy with your current solution =)